framework/src/application/node/forger/strategies.js

```diff
@@ -0,0 +1,118 @@
+ */
+
+'use strict';
+
+const { MinHeap, MaxHeap } = require('@liskhq/lisk-transaction-pool');
+const { Status: TransactionStatus } = require('@liskhq/lisk-transactions');
+
+class HighFeeForgingStrategy {
+	constructor({
+		// components
+		logger,
+		// Modules
+		chainModule,
+		transactionPoolModule,
+		// constants
+		maxPayloadLength,
+	}) {
+		this.chainModule = chainModule;
+		this.transactionPoolModule = transactionPoolModule;
+		this.logger = logger;
+
+		this.constants = { maxPayloadLength: BigInt(maxPayloadLength) };
+	}
+
+	async getTransactionsForBlock() {
+		// Initialize array to select transactions
+		const readyTransactions = [];
+
+		// Initialize block size with 0
+		let blockPayloadSize = BigInt(0);
+
+		// Initialize state store which will be discarded after selection
+		const stateStore = this.chainModule.newStateStore();
+
+		// Get processable transactions from transaction pool
+		const processableTransactions = this.transactionPoolModule.getProcessableTransactions();
+
+		// Convert sender transactions to min heap by nonce
+		const transactionsBySender = Object.keys(processableTransactions).reduce(
+			(obj, senderId) => {
+				const minHeapByNonce = new MinHeap();
+				processableTransactions[senderId].forEach(t =>
+					minHeapByNonce.push(t.nonce, t),
+				);
+				obj[senderId] = minHeapByNonce;
+
+				return obj;
+			},
+			{},
+		);
+
+		// Loop till we have last account exhausted to pick transactions
+		while (Object.keys(transactionsBySender).length !== 0) {
+			const feePriorityHeap = new MaxHeap();
+
+			// Prepare max heap for high fee priority for lowest nonce of all available accounts
+			for (const senderId of Object.keys(transactionsBySender)) {
+				const lowestNonceTrx = transactionsBySender[senderId].peek().value;
+				feePriorityHeap.push(lowestNonceTrx.fee, lowestNonceTrx);
+			}
+
+			// Get the transaction with highest fee and lowest nonce
+			const lowestNonceHighestFeeTrx = feePriorityHeap.pop().value;
+
+			// Try to process transaction
+			const result = await this.chainModule.processTransactionsWithStateStore(
+				[lowestNonceHighestFeeTrx],
+				stateStore,
+			);
+
+			// If transaction can't be processed then discard all transactions
+			// from that account as other transactions will be higher nonce
+			if (result.transactionsResponses[0].status !== TransactionStatus.OK) {
+				delete transactionsBySender[lowestNonceHighestFeeTrx.senderId];
+
+				// eslint-disable-next-line no-continue
+				continue;
+			}
+
+			// If transaction byte size can't fit in max payload length
+			// then discard all transactions from that account as
+			// other transactions will be higher nonce
+			const trsByteSize = BigInt(lowestNonceHighestFeeTrx.getBytes().length);
+			if (blockPayloadSize + trsByteSize > this.constants.maxPayloadLength) {
+				delete transactionsBySender[lowestNonceHighestFeeTrx.senderId];
+				// eslint-disable-next-line no-continue
+				continue;
```

Comment:
i think in this case, you can just return all the transaction you have already

Reply:
There can be any transaction from another account that can fit in the available space.

Reply:
that's true, but i wasnt sure how feasible it is that to happen. in case we leave 1 byte, then we will go through all of the transaction to see if it fits
that's why i thought it's safer to exit here than continuing

Reply:
We can add a threshold limit, but the original idea of the task was to maximize the forging fee. So I tried to fill in the block every possible way.

Reply:
I agree that it should maximize the fee, but it would be too time consuming to go through and apply all transactions, in the worst case scenario it would be 4k transactions